ddlitlab2024/dataset/pytorch.py

```diff
@@ -178,25 +194,36 @@ def query_image_data(
         stamps = []
         image_data = []
 
+        # Define the preprocessing pipeline
+        preprocessing = v2.Compose(
+            [
+                v2.ToImage(),
+                v2.ToDtype(torch.float32, scale=True),
+                v2.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+            ]
+        )
+
         # Get the raw image data
         for stamp, data in response:
             # Deserialize the image data
             image = np.frombuffer(data, dtype=np.uint8).reshape(480, 480, 3)
-            # Make chw from hwc
-            image = np.moveaxis(image, -1, 0)
+            # Resize the image
+            image = cv2.resize(image, (resolution, resolution), interpolation=cv2.INTER_AREA)
```

Comment:
Do we do this resize for faster training / less VRAM usage? I assume in our DB all images are still 480x480 or did that change for the NAO images?

Reply:
I added this so we can train using a lower resolution. Some models are e.g. pretrained with 224*224 so it is best that we use this resolution. It also helps with vram. The resolution in the DB would serve as our maximum resolution.